npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb

```diff
@@ -64,6 +65,22 @@ def instrument_package_manager_version
         )
       end
 
+      def yarn_version
+        @yarn_version ||= begin
+          package = JSON.load(package_json.content)
+          if (pkgmanager = package.fetch("packageManager"))
+            @yarn_version = get_yarn_version_from_path(pkgmanager)
+          elsif yarn_lock
+            @yarn_version = 1
+          end
+        end
+      end
+
+      def get_yarn_version_from_path(path)
+        version_match = path.match(/yarn@(?<version>\d+.\d+.\d+)/)
+        version_match.named_captures.fetch("version")
```

Comment:
Are we guaranteed to have `path` be a String that matches this pattern? If not, we'll likely end up with some `NoMethodError`s on these two lines.